In [23]:
%matplotlib inline
import pandas as pd
import numpy as np

In [33]:
keywords = pd.read_csv('keywords.csv')
sites = pd.read_csv('sites.csv')
rankings = pd.read_csv('rankings.csv')

rankings = rankings.merge(keywords, on='id_keyword')
rankings = rankings.merge(sites, on='id_site')
rankings = rankings[['rank','keyword', 'domain']]
rankings = rankings.drop_duplicates()  # Note the data is dirty and contains about 10000 duplicates
rankings.head()

rank                                         keyword     domain
0    43                  bachelor's degree in Economics  www.au.dk
1    42                  bachelor's degree in Economics  www.au.dk
2    91           bachelor's degree in English Language  www.au.dk
3    29                bachelor's degree in Archaeology  www.au.dk
4     7  bachelor's degree in Business & Commercial Law  www.au.dk

Each entry represents an appeareance of the given university domain amoung the first 100 Google search results obtained with the corresponding query (keyword). The rank is the position the university appeared on the results page.

In [32]:
rankings.domain.describe()

count           24491
unique           1448
top       www.asu.edu
freq              223
Name: domain, dtype: object

In [29]:
sites.count()

id_site    9260
domain     9258
dtype: int64

We see that 1448 / 9260 of the universities considered appear at least once in the google search results.
With unweighted results we see the ase.edu (Arizona state university) appears the most (for 223 keywords out of a possible 339)

We can see which universities are the most present in Google search results simply by seeing on how many queries they appear (see following table).

Next steps will be:
- weight the result with the rank attribute
- weight the keywords with search volume and cpc
- divide rankings into categories (degree type, subject, country etc..)

In [36]:
rankings.domain.value_counts().head()

www.asu.edu         223
www.bu.edu          209
www.nyu.edu         194
www.stanford.edu    180
www.psu.edu         174
Name: domain, dtype: int64

In [37]:
rankings.domain.value_counts().to_csv('unweight_ranking.csv')